packages/bson-transpilers/symbols/go/templates.yaml

```diff
@@ -0,0 +1,912 @@
+    DoubleSymbolTemplate: &DoubleSymbolTemplate !!js/function >
+        () => ''
+    DoubleSymbolArgsTemplate: &DoubleSymbolArgsTemplate !!js/function >
+        (lhs, arg, type) => {
+            if (!arg)
+                arg = 0;
+            switch(type) {
+                case '_string':
+                    this.declarations.addFunc([]
+                        .concat('var parseFloat64 = func(str string) float64 {')
+                        .concat('  f64, err := strconv.ParseFloat(str, 64)')
+                        .concat('  if err != nil {')
+                        .concat('    log.Fatal(err)')
+                        .concat('  }')
+                        .concat('  return f64')
+                        .concat('}')
+                        .join('\n'));
+                    return `parseFloat64(${arg})`
+                default:
+                    return `float64(${arg})`;
+            }
+        }
+    Int32SymbolTemplate: &Int32SymbolTemplate !!js/function >
+        () => ''
+    Int32SymbolArgsTemplate: &Int32SymbolArgsTemplate !!js/function >
+        (lhs, arg, type) => {
+            if (!arg)
+                arg = 0
+            switch(type) {
+                case '_string':
+                    this.declarations.addFunc([]
+                        .concat('var parseInt32 = func(str string) int32 {')
+                        .concat('i64, err := strconv.ParseInt(str, 10, 32)')
+                        .concat('  if err != nil {')
+                        .concat('    log.Fatal(err)')
+                        .concat('  }')
+                        .concat('  return int32(i64)')
+                        .concat('}')
+                        .join('\n'));
+                    return `parseInt32(${arg})`;
+                default:
+                    return `int32(${arg})`;
+            }
+        }
+    LongSymbolTemplate: &LongSymbolTemplate !!js/function >
+        () => ''
+    LongSymbolArgsTemplate: &LongSymbolArgsTemplate !!js/function >
+        (lhs, arg, type) => {
+            if (!arg)
+                arg = 0
+            switch(type) {
+                case '_string':
+                    this.declarations.addFunc([]
+                        .concat('var parseInt = func(str string) int64 {')
+                        .concat('  i64, err := strconv.ParseInt(str, 10, 64)')
+                        .concat('  if err != nil {')
+                        .concat('    log.Fatal(err)')
+                        .concat('  }')
+                        .concat('  return i64')
+                        .concat('}')
+                        .join('\n'));
+                    return `parseInt64(${arg})`;
+                default:
+                    return `int64(${arg})`;
+            }
+        }
+    RegExpSymbolTemplate: &RegExpSymbolTemplate !!js/function >
+        () => 'regex'
+    RegExpSymbolArgsTemplate: &RegExpSymbolArgsTemplate null # Args: lhs, pattern, flags
```

Comment:
I think there may be a bug handling the `RegExp` function. If you write a query like:
```
{x: RegExp('a', '')}
```

The generated Go code looks like:
```go
bson.D{{"x", RegExp}}
```
which doesn't compile and seems to be missing the arguments.

For comparison, the Python code looks like:
```python
{
    'x': re.compile(r"a")
}
```